trompace/client/client_itemlist.py

```diff
@@ -0,0 +1,128 @@
+                         description = description, position = position):
+    """
+    """
+    mutation = mutation_create_listitem(contributor = contributor, 
+                                        name = name,
+                                        description = value, 
+                                        position = position)
+
+    resp = trompace.connection.submit_query(mutation)
+    result = resp.get("data", {}).get("CreateListItem")
+
+    if result:
+        element_id = resp["data"]["CreateItemList"]["identifier"]
+    else:
+        # Raise Error
+        pass
+
+    return element_id
+
+
+def merge_itemlist_itemlistelement_nodes(itemlist_id = itemlist_id, 
+                                         element_id = element_id):
+    """
+    """
+    mutation = mutation_add_itemlist_itemlist_element(itemlist_id = itemlist_id, 
+                                                      element_id = element_id)
+    resp = trompace.connection.submit_query(mutation)
+    result = resp.get("data", {}).get("MergeItemListItemListElement")
+
+    if not result:
+        # Raise Error
+        pass
+
+
+def merge_listitem_nextitem_nodes(listitem_id = previousItemId, 
+                                  nextitem_id = element_id):
+    """
+    """
+    mutation = mutation_add_listitem_nextitem(listitem_id = previousItemId, 
+                                              nextitem_id = element_id)
+    resp = trompace.connection.submit_query(mutation)
+    result = resp.get("data", {}).get("MergeListItemNextItem")
+
+    if not result:
+        # Raise Error
+        pass
+
+
+def merge_listitem_item_nodes(listitem_id = listitem_id, item_id = item_id):
+    """
+    """
+    mutation =  mutation_add_listitem_item(listitem_id = listitem_id, 
+                                           item_id = item_id)
+
+    resp = trompace.connection.submit_query(mutation)
+    result = resp.get("data", {}).get("MergeListItemItem")
+
+    if not result:
+        # Raise Error
+        pass
+
+
+def create_itemlist(contributor: str, name: str, description: str, 
+                    node_ids: list = None, values: list = None, ordered: bool):
+    """
+    Main function
+    """
+    if values:
+        listitems = values
+    elif node_ids:
+        listitems = node_ids
+
+    itemlist_id = create_itemlist_node(contributor = contributor, name = name, 
+                                       description = description, ordered = ordered):
+
+    position = 0
+    previous_item_id = None
+    for item in listitems:
+        if values:
+            description = item
+
+        listitem_id = create_listitem_node(contributor = contributor, name = name, 
+                                           description = description, position = position):
+
+        merge_itemlist_itemlistelement_nodes(itemlist_id = itemlist_id, element_id = listitem_id)
+
+        if node_ids:
+            merge_listitem_item_nodes(listitem_id = listitem_id, item_id = item)
```

Comment:
I'm thinking out loud here - I wonder if we should add a check here to ensure that all node ids exist in the CE before performing this link. Otherwise this could randomly fail half-way through creating the list.

Reply:
A check can be done using a query to search within the GraphQL db?